fix: Add npm fallback when bun install fails with alias packages

## Problem

Bun has issues resolving npm alias packages (`*-cjs` packages) that are referenced in `package-lock.json`. This causes installation failures with 404 errors:

```
📦 Installing dependencies with Bun...
error: GET https://registry.npmjs.org/string-width-cjs/-/string-width-cjs-4.2.3.tgz - 404
error: GET https://registry.npmjs.org/strip-ansi-cjs/-/strip-ansi-cjs-6.0.1.tgz - 404
error: GET https://registry.npmjs.org/wrap-ansi-cjs/-/wrap-ansi-cjs-7.0.0.tgz - 404
❌ Installation failed: Command failed: bun install --force
```

This also leads to PM2 errors when the worker tries to start:
```
Error: Cannot find module '.../node_modules/pm2/lib/ProcessContainerFork.js'
```

## Root Cause

These `*-cjs` packages are npm aliases (defined in `package-lock.json`), not real packages on npm. Bun 1.3.x cannot resolve these aliases correctly when migrating from a `package-lock.json`.

## Proposed Fix

Add an npm fallback to `plugin/scripts/smart-install.js` when bun install fails:

```javascript
/**
 * Install dependencies using Bun with npm fallback
 *
 * Bun has issues with npm alias packages (e.g., string-width-cjs, strip-ansi-cjs)
 * that are defined in package-lock.json. When bun fails with 404 errors for these
 * packages, we fall back to npm which handles aliases correctly.
 */
function installDeps() {
  const bunPath = getBunPath();
  if (!bunPath) {
    throw new Error('Bun executable not found');
  }

  console.error('📦 Installing dependencies with Bun...');
  const bunCmd = IS_WINDOWS && bunPath.includes(' ') ? `"${bunPath}"` : bunPath;

  let bunSucceeded = false;
  try {
    execSync(`${bunCmd} install`, { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
    bunSucceeded = true;
  } catch {
    try {
      execSync(`${bunCmd} install --force`, { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
      bunSucceeded = true;
    } catch {
      // Bun failed completely, will try npm fallback
    }
  }

  // Fallback to npm if bun failed (handles npm alias packages correctly)
  if (!bunSucceeded) {
    console.error('⚠️  Bun install failed, falling back to npm...');
    console.error('   (This can happen with npm alias packages like *-cjs)');
    try {
      execSync('npm install', { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
    } catch (npmError) {
      throw new Error(`Both bun and npm install failed: ${npmError.message}`);
    }
  }

  // Write version marker...
}
```

## Workaround

Until this is fixed, users can manually run:
```bash
cd ~/.claude/plugins/marketplaces/thedotmack
npm install
```

## Environment

- Bun: 1.3.4
- OS: Linux (also reported on macOS)
- claude-mem: 7.1.3

## Related

- #261 (closed but issue persists)
- #253 (PM2 MODULE_NOT_FOUND)

--- a/plugin/scripts/smart-install.js
+++ b/plugin/scripts/smart-install.js
@@ -5,30 +5,24 @@
  * Ensures Bun runtime and uv (Python package manager) are installed
  * (auto-installs if missing) and handles dependency installation when needed.
  */
-import { existsSync, readFileSync, writeFileSync } from "fs";
-import { execSync, spawnSync } from "child_process";
-import { join } from "path";
-import { homedir } from "os";
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { execSync, spawnSync } from 'child_process';
+import { join } from 'path';
+import { homedir } from 'os';
 
-const ROOT = join(
-  homedir(),
-  ".claude",
-  "plugins",
-  "marketplaces",
-  "thedotmack",
-);
-const MARKER = join(ROOT, ".install-version");
-const IS_WINDOWS = process.platform === "win32";
+const ROOT = join(homedir(), '.claude', 'plugins', 'marketplaces', 'thedotmack');
+const MARKER = join(ROOT, '.install-version');
+const IS_WINDOWS = process.platform === 'win32';
 
 /**
  * Check if Bun is installed and accessible
  */
 function isBunInstalled() {
   try {
-    const result = spawnSync("bun", ["--version"], {
-      encoding: "utf-8",
-      stdio: ["pipe", "pipe", "pipe"],
-      shell: IS_WINDOWS,
+    const result = spawnSync('bun', ['--version'], {
+      encoding: 'utf-8',
+      stdio: ['pipe', 'pipe', 'pipe'],
+      shell: IS_WINDOWS
     });
     if (result.status === 0) return true;
   } catch {
@@ -37,8 +31,8 @@ function isBunInstalled() {
 
   // Check common installation paths (handles fresh installs before PATH reload)
   const bunPaths = IS_WINDOWS
-    ? [join(homedir(), ".bun", "bin", "bun.exe")]
-    : [join(homedir(), ".bun", "bin", "bun"), "/usr/local/bin/bun"];
+    ? [join(homedir(), '.bun', 'bin', 'bun.exe')]
+    : [join(homedir(), '.bun', 'bin', 'bun'), '/usr/local/bin/bun'];
 
   return bunPaths.some(existsSync);
 }
@@ -49,20 +43,20 @@ function isBunInstalled() {
 function getBunPath() {
   // Try PATH first
   try {
-    const result = spawnSync("bun", ["--version"], {
-      encoding: "utf-8",
-      stdio: ["pipe", "pipe", "pipe"],
-      shell: IS_WINDOWS,
+    const result = spawnSync('bun', ['--version'], {
+      encoding: 'utf-8',
+      stdio: ['pipe', 'pipe', 'pipe'],
+      shell: IS_WINDOWS
     });
-    if (result.status === 0) return "bun";
+    if (result.status === 0) return 'bun';
   } catch {
     // Not in PATH
   }
 
   // Check common installation paths
   const bunPaths = IS_WINDOWS
-    ? [join(homedir(), ".bun", "bin", "bun.exe")]
-    : [join(homedir(), ".bun", "bin", "bun"), "/usr/local/bin/bun"];
+    ? [join(homedir(), '.bun', 'bin', 'bun.exe')]
+    : [join(homedir(), '.bun', 'bin', 'bun'), '/usr/local/bin/bun'];
 
   for (const bunPath of bunPaths) {
     if (existsSync(bunPath)) return bunPath;
@@ -79,10 +73,10 @@ function getBunVersion() {
   if (!bunPath) return null;
 
   try {
-    const result = spawnSync(bunPath, ["--version"], {
-      encoding: "utf-8",
-      stdio: ["pipe", "pipe", "pipe"],
-      shell: IS_WINDOWS,
+    const result = spawnSync(bunPath, ['--version'], {
+      encoding: 'utf-8',
+      stdio: ['pipe', 'pipe', 'pipe'],
+      shell: IS_WINDOWS
     });
     return result.status === 0 ? result.stdout.trim() : null;
   } catch {
@@ -95,10 +89,10 @@ function getBunVersion() {
  */
 function isUvInstalled() {
   try {
-    const result = spawnSync("uv", ["--version"], {
-      encoding: "utf-8",
-      stdio: ["pipe", "pipe", "pipe"],
-      shell: IS_WINDOWS,
+    const result = spawnSync('uv', ['--version'], {
+      encoding: 'utf-8',
+      stdio: ['pipe', 'pipe', 'pipe'],
+      shell: IS_WINDOWS
     });
     if (result.status === 0) return true;
   } catch {
@@ -107,15 +101,8 @@ function isUvInstalled() {
 
   // Check common installation paths (handles fresh installs before PATH reload)
   const uvPaths = IS_WINDOWS
-    ? [
-        join(homedir(), ".local", "bin", "uv.exe"),
-        join(homedir(), ".cargo", "bin", "uv.exe"),
-      ]
-    : [
-        join(homedir(), ".local", "bin", "uv"),
-        join(homedir(), ".cargo", "bin", "uv"),
-        "/usr/local/bin/uv",
-      ];
+    ? [join(homedir(), '.local', 'bin', 'uv.exe'), join(homedir(), '.cargo', 'bin', 'uv.exe')]
+    : [join(homedir(), '.local', 'bin', 'uv'), join(homedir(), '.cargo', 'bin', 'uv'), '/usr/local/bin/uv'];
 
   return uvPaths.some(existsSync);
 }
@@ -125,10 +112,10 @@ function isUvInstalled() {
  */
 function getUvVersion() {
   try {
-    const result = spawnSync("uv", ["--version"], {
-      encoding: "utf-8",
-      stdio: ["pipe", "pipe", "pipe"],
-      shell: IS_WINDOWS,
+    const result = spawnSync('uv', ['--version'], {
+      encoding: 'utf-8',
+      stdio: ['pipe', 'pipe', 'pipe'],
+      shell: IS_WINDOWS
     });
     return result.status === 0 ? result.stdout.trim() : null;
   } catch {
@@ -140,22 +127,22 @@ function getUvVersion() {
  * Install Bun automatically based on platform
  */
 function installBun() {
-  console.error("🔧 Bun not found. Installing Bun runtime...");
+  console.error('🔧 Bun not found. Installing Bun runtime...');
 
   try {
     if (IS_WINDOWS) {
       // Windows: Use PowerShell installer
-      console.error("   Installing via PowerShell...");
+      console.error('   Installing via PowerShell...');
       execSync('powershell -c "irm bun.sh/install.ps1 | iex"', {
-        stdio: "inherit",
-        shell: true,
+        stdio: 'inherit',
+        shell: true
       });
     } else {
       // Unix/macOS: Use curl installer
-      console.error("   Installing via curl...");
-      execSync("curl -fsSL https://bun.sh/install | bash", {
-        stdio: "inherit",
-        shell: true,
+      console.error('   Installing via curl...');
+      execSync('curl -fsSL https://bun.sh/install | bash', {
+        stdio: 'inherit',
+        shell: true
       });
     }
 
@@ -168,17 +155,15 @@ function installBun() {
       // Bun may be installed but not in PATH yet for this session
       // Try common installation paths
       const bunPaths = IS_WINDOWS
-        ? [join(homedir(), ".bun", "bin", "bun.exe")]
-        : [join(homedir(), ".bun", "bin", "bun"), "/usr/local/bin/bun"];
+        ? [join(homedir(), '.bun', 'bin', 'bun.exe')]
+        : [join(homedir(), '.bun', 'bin', 'bun'), '/usr/local/bin/bun'];
 
       for (const bunPath of bunPaths) {
         if (existsSync(bunPath)) {
           console.error(`✅ Bun installed at ${bunPath}`);
-          console.error("⚠️  Please restart your terminal or add Bun to PATH:");
+          console.error('⚠️  Please restart your terminal or add Bun to PATH:');
           if (IS_WINDOWS) {
-            console.error(
-              `   $env:Path += ";${join(homedir(), ".bun", "bin")}"`,
-            );
+            console.error(`   $env:Path += ";${join(homedir(), '.bun', 'bin')}"`);
           } else {
             console.error(`   export PATH="$HOME/.bun/bin:$PATH"`);
           }
@@ -186,19 +171,19 @@ function installBun() {
         }
       }
 
-      throw new Error("Bun installation completed but binary not found");
+      throw new Error('Bun installation completed but binary not found');
     }
   } catch (error) {
-    console.error("❌ Failed to install Bun automatically");
-    console.error("   Please install manually:");
+    console.error('❌ Failed to install Bun automatically');
+    console.error('   Please install manually:');
     if (IS_WINDOWS) {
-      console.error("   - winget install Oven-sh.Bun");
+      console.error('   - winget install Oven-sh.Bun');
       console.error('   - Or: powershell -c "irm bun.sh/install.ps1 | iex"');
     } else {
-      console.error("   - curl -fsSL https://bun.sh/install | bash");
-      console.error("   - Or: brew install oven-sh/bun/bun");
+      console.error('   - curl -fsSL https://bun.sh/install | bash');
+      console.error('   - Or: brew install oven-sh/bun/bun');
     }
-    console.error("   Then restart your terminal and try again.");
+    console.error('   Then restart your terminal and try again.');
     throw error;
   }
 }
@@ -207,25 +192,22 @@ function installBun() {
  * Install uv automatically based on platform
  */
 function installUv() {
-  console.error("🐍 Installing uv for Python/Chroma support...");
+  console.error('🐍 Installing uv for Python/Chroma support...');
 
   try {
     if (IS_WINDOWS) {
       // Windows: Use PowerShell installer
-      console.error("   Installing via PowerShell...");
-      execSync(
-        'powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"',
-        {
-          stdio: "inherit",
-          shell: true,
-        },
-      );
+      console.error('   Installing via PowerShell...');
+      execSync('powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"', {
+        stdio: 'inherit',
+        shell: true
+      });
     } else {
       // Unix/macOS: Use curl installer
-      console.error("   Installing via curl...");
-      execSync("curl -LsSf https://astral.sh/uv/install.sh | sh", {
-        stdio: "inherit",
-        shell: true,
+      console.error('   Installing via curl...');
+      execSync('curl -LsSf https://astral.sh/uv/install.sh | sh', {
+        stdio: 'inherit',
+        shell: true
       });
     }
 
@@ -238,24 +220,15 @@ function installUv() {
       // uv may be installed but not in PATH yet for this session
       // Try common installation paths
       const uvPaths = IS_WINDOWS
-        ? [
-            join(homedir(), ".local", "bin", "uv.exe"),
-            join(homedir(), ".cargo", "bin", "uv.exe"),
-          ]
-        : [
-            join(homedir(), ".local", "bin", "uv"),
-            join(homedir(), ".cargo", "bin", "uv"),
-            "/usr/local/bin/uv",
-          ];
+        ? [join(homedir(), '.local', 'bin', 'uv.exe'), join(homedir(), '.cargo', 'bin', 'uv.exe')]
+        : [join(homedir(), '.local', 'bin', 'uv'), join(homedir(), '.cargo', 'bin', 'uv'), '/usr/local/bin/uv'];
 
       for (const uvPath of uvPaths) {
         if (existsSync(uvPath)) {
           console.error(`✅ uv installed at ${uvPath}`);
-          console.error("⚠️  Please restart your terminal or add uv to PATH:");
+          console.error('⚠️  Please restart your terminal or add uv to PATH:');
           if (IS_WINDOWS) {
-            console.error(
-              `   $env:Path += ";${join(homedir(), ".local", "bin")}"`,
-            );
+            console.error(`   $env:Path += ";${join(homedir(), '.local', 'bin')}"`);
           } else {
             console.error(`   export PATH="$HOME/.local/bin:$PATH"`);
           }
@@ -263,21 +236,19 @@ function installUv() {
         }
       }
 
-      throw new Error("uv installation completed but binary not found");
+      throw new Error('uv installation completed but binary not found');
     }
   } catch (error) {
-    console.error("❌ Failed to install uv automatically");
-    console.error("   Please install manually:");
+    console.error('❌ Failed to install uv automatically');
+    console.error('   Please install manually:');
     if (IS_WINDOWS) {
-      console.error("   - winget install astral-sh.uv");
-      console.error(
-        '   - Or: powershell -c "irm https://astral.sh/uv/install.ps1 | iex"',
-      );
+      console.error('   - winget install astral-sh.uv');
+      console.error('   - Or: powershell -c "irm https://astral.sh/uv/install.ps1 | iex"');
     } else {
-      console.error("   - curl -LsSf https://astral.sh/uv/install.sh | sh");
-      console.error("   - Or: brew install uv (macOS)");
+      console.error('   - curl -LsSf https://astral.sh/uv/install.sh | sh');
+      console.error('   - Or: brew install uv (macOS)');
     }
-    console.error("   Then restart your terminal and try again.");
+    console.error('   Then restart your terminal and try again.');
     throw error;
   }
 }
@@ -286,10 +257,10 @@ function installUv() {
  * Check if dependencies need to be installed
  */
 function needsInstall() {
-  if (!existsSync(join(ROOT, "node_modules"))) return true;
+  if (!existsSync(join(ROOT, 'node_modules'))) return true;
   try {
-    const pkg = JSON.parse(readFileSync(join(ROOT, "package.json"), "utf-8"));
-    const marker = JSON.parse(readFileSync(MARKER, "utf-8"));
+    const pkg = JSON.parse(readFileSync(join(ROOT, 'package.json'), 'utf-8'));
+    const marker = JSON.parse(readFileSync(MARKER, 'utf-8'));
     return pkg.version !== marker.version || getBunVersion() !== marker.bun;
   } catch {
     return true;
@@ -306,30 +277,22 @@ function needsInstall() {
 function installDeps() {
   const bunPath = getBunPath();
   if (!bunPath) {
-    throw new Error("Bun executable not found");
+    throw new Error('Bun executable not found');
   }
 
-  console.error("📦 Installing dependencies with Bun...");
+  console.error('📦 Installing dependencies with Bun...');
 
   // Quote path for Windows paths with spaces
-  const bunCmd = IS_WINDOWS && bunPath.includes(" ") ? `"${bunPath}"` : bunPath;
+  const bunCmd = IS_WINDOWS && bunPath.includes(' ') ? `"${bunPath}"` : bunPath;
 
   let bunSucceeded = false;
   try {
-    execSync(`${bunCmd} install`, {
-      cwd: ROOT,
-      stdio: "inherit",
-      shell: IS_WINDOWS,
-    });
+    execSync(`${bunCmd} install`, { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
     bunSucceeded = true;
   } catch {
     // First attempt failed, try with force flag
     try {
-      execSync(`${bunCmd} install --force`, {
-        cwd: ROOT,
-        stdio: "inherit",
-        shell: IS_WINDOWS,
-      });
+      execSync(`${bunCmd} install --force`, { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
       bunSucceeded = true;
     } catch {
       // Bun failed completely, will try npm fallback
@@ -338,30 +301,23 @@ function installDeps() {
 
   // Fallback to npm if bun failed (handles npm alias packages correctly)
   if (!bunSucceeded) {
-    console.error("⚠️  Bun install failed, falling back to npm...");
-    console.error("   (This can happen with npm alias packages like *-cjs)");
+    console.error('⚠️  Bun install failed, falling back to npm...');
+    console.error('   (This can happen with npm alias packages like *-cjs)');
     try {
-      execSync("npm install", {
-        cwd: ROOT,
-        stdio: "inherit",
-        shell: IS_WINDOWS,
-      });
+      execSync('npm install', { cwd: ROOT, stdio: 'inherit', shell: IS_WINDOWS });
     } catch (npmError) {
-      throw new Error(`Both bun and npm install failed: ${npmError.message}`);
+      throw new Error('Both bun and npm install failed: ' + npmError.message);
     }
   }
 
   // Write version marker
-  const pkg = JSON.parse(readFileSync(join(ROOT, "package.json"), "utf-8"));
-  writeFileSync(
-    MARKER,
-    JSON.stringify({
-      version: pkg.version,
-      bun: getBunVersion(),
-      uv: getUvVersion(),
-      installedAt: new Date().toISOString(),
-    }),
-  );
+  const pkg = JSON.parse(readFileSync(join(ROOT, 'package.json'), 'utf-8'));
+  writeFileSync(MARKER, JSON.stringify({
+    version: pkg.version,
+    bun: getBunVersion(),
+    uv: getUvVersion(),
+    installedAt: new Date().toISOString()
+  }));
 }
 
 // Main execution
@@ -372,8 +328,8 @@ try {
 
     // Re-check after installation
     if (!isBunInstalled()) {
-      console.error("❌ Bun is required but not available in PATH");
-      console.error("   Please restart your terminal after installation");
+      console.error('❌ Bun is required but not available in PATH');
+      console.error('   Please restart your terminal after installation');
       process.exit(1);
     }
   }
@@ -384,8 +340,8 @@ try {
 
     // Re-check after installation
     if (!isUvInstalled()) {
-      console.error("❌ uv is required but not available in PATH");
-      console.error("   Please restart your terminal after installation");
+      console.error('❌ uv is required but not available in PATH');
+      console.error('   Please restart your terminal after installation');
       process.exit(1);
     }
   }
@@ -393,9 +349,9 @@ try {
   // Step 3: Install dependencies if needed
   if (needsInstall()) {
     installDeps();
-    console.error("✅ Dependencies installed");
+    console.error('✅ Dependencies installed');
   }
 } catch (e) {
-  console.error("❌ Installation failed:", e.message);
+  console.error('❌ Installation failed:', e.message);
   process.exit(1);
 }

